org.eclipse.m2e.jdt/src/org/eclipse/m2e/jdt/internal/AbstractJavaProjectConfigurator.java

```diff
@@ -363,6 +368,8 @@ protected void addProjectSourceFolders(IClasspathDescriptor classpath, Map<Strin
       for(MojoExecution resources : projectFacade.getMojoExecutions(RESOURCES_PLUGIN_GROUP_ID,
           RESOURCES_PLUGIN_ARTIFACT_ID, mon.newChild(1), GOAL_TESTRESOURCES)) {
         testResourcesEncoding = maven.getMojoParameterValue(mavenProject, resources, "encoding", String.class, monitor); //$NON-NLS-1$
+        isTestResourcesSkipped = "true" //$NON-NLS-1$
+            .equals(maven.getMojoParameterValue(mavenProject, resources, "skip", String.class, monitor)); //$NON-NLS-1$
```

Comment:
```suggestion
        isTestResourcesSkipped = Boolean.TRUE
            .equals(maven.getMojoParameterValue(mavenProject, resources, "skip", Boolean.class, monitor)); //$NON-NLS-1$
```

The skip field of the resources-plugin is of type `boolean`:
https://github.com/apache/maven-resources-plugin/blob/6fce82eb63d03d7efdc993b2e222952507039b28/src/main/java/org/apache/maven/plugins/resources/TestResourcesMojo.java#L60-L61

Therefore I think it is better to let maven handle the value conversion.